server/src/main/java/org/opensearch/rest/pagination/PaginationStrategy.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+import java.util.List;
+
+/**
+ * Interface to be implemented by any strategy getting used for paginating rest responses.
+ *
+ * @opensearch.internal
+ */
+public interface PaginationStrategy<T> {
```

Comment:
Might be good to name this as PageRequestOption / PageTokenOption - which takes input from Request and ClusterState to come up with pageToken. I think the `List<T> getPageElements` can remain outside given that PageOptions are translated to filters.

Reply:
Updated in the latest revision.